sys/xtimer/xtimer.c

```diff
@@ -32,6 +32,12 @@
 #define ENABLE_DEBUG 0
 #include "debug.h"
 
+typedef struct {
+    mutex_t *mutex;
+    thread_t *thread;
+    int timeout;
```

Comment:
When the mutex times out, this flag is set in order to indicate whether it timed out or not (see L257).